pkg/registry/ingress/strategy.go

```diff
@@ -114,3 +116,20 @@ func MatchIngress(label labels.Selector, field fields.Selector) generic.Matcher
 		},
 	}
 }
+
+type ingressStatusStrategy struct {
+	ingressStrategy
+}
+
+var StatusStrategy = ingressStatusStrategy{Strategy}
+
+func (ingressStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newIngress := obj.(*extensions.Ingress)
+	oldIngress := old.(*extensions.Ingress)
+	// status changes are not allowed to update spec
+	newIngress.Spec = oldIngress.Spec
+}
+
+func (ingressStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
```

Comment:
I think we are inconsistent here, but in general, if we are doing a status update, we should be safe in calling validateStatusUpdate(), strategy_test covers that you cannot update spec as part of status, so at best you are catching object.metadata changes which are accounted for in the validateStatusUpdate call pattern today.